Jetpack E2E: miscellaneous block flows from the important tasks.

worldomonation opened the pull request:
Related to: https://github.com/Automattic/wp-calypso/issues/80730.

## Proposed Changes

This PR adds two new specs under the miscellaneous blocks:
- related posts
- markdown

## Testing Instructions

Ensure the following build configurations are passing:
  - [x] Calypso E2E (desktop)
  - [x] Jetpack E2E Simple (mobile)
  - [x] Jetpack E2E Simple (desktop)
  - [x] Jetpack E2E AT (desktop)

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

worldomonation committed:
packages/calypso-e2e/src/lib/blocks/block-flows/markdown.ts
packages/calypso-e2e/src/lib/blocks/block-flows/index.ts
- add new block flow for Markdown.

test/e2e/specs/blocks/blocks__jetpack-writing.ts
- add new block flow

worldomonation committed:
packages/calypso-e2e/src/lib/components/editor-block-toolbar-component.ts
- narrow selector

worldomonation committed:
packages/calypso-e2e/src/lib/blocks/block-flows/index.ts
packages/calypso-e2e/src/lib/blocks/block-flows/related-posts.ts
- add new POM for the Related Posts block.

test/e2e/specs/blocks/blocks__jetpack-other.ts
- add new flow

worldomonation committed:
packages/calypso-e2e/src/lib/components/editor-block-toolbar-component.ts
- assign to locator

worldomonation commented on packages/calypso-e2e/src/lib/blocks/block-flows/related-posts.ts:
```diff
@@ -0,0 +1,57 @@
+		const noRelatedPostsNotice = await block.getByText( /Preview unavailable/ ).count();
+		const postSlotCount = await block.getByRole( 'menuitem' ).count();
+
+		if ( noRelatedPostsNotice === postSlotCount ) {
+			this.noRelatedPosts = true;
+		}
```
This is hacky and could stop working at anytime, but we don't really have a better way to test whether the block renders on the published page without having some sort of heuristic on when the block is expected to appear.

dpasque replied:
Ah sure, this is a tricky situation! I think this is a good place to start fwiw 🙂

worldomonation committed:
packages/calypso-e2e/src/lib/blocks/block-flows/related-posts.ts
- add ability to specify header.
- update validator method to also check for header if specified.

test/e2e/specs/blocks/blocks__jetpack-other.ts
- if/else clause for the Related Blocks spec.

test/e2e/specs/blocks/shared/block-smoke-testing.ts
- add step to write the title.

worldomonation committed:
packages/calypso-e2e/src/lib/blocks/block-flows/markdown.ts
- fix JSDoc

packages/calypso-e2e/src/lib/blocks/block-flows/related-posts.ts
- fix JSDoc
- add explanation on how to turn on the block.

test/e2e/specs/blocks/blocks__jetpack-other.ts
- remove gating

worldomonation commented on test/e2e/specs/blocks/shared/block-smoke-testing.ts:
```diff
@@ -48,6 +48,10 @@ export function createBlockTests( specName: string, blockFlows: BlockFlow[] ): v
 			await editorPage.visit( 'post' );
 		} );
 
+		it( 'Enter post title', async () => {
+			await editorPage.enterTitle( `${ specName } - ${ DataHelper.getDateString( 'ISO-8601' ) }` );
+		} );
+
```
This is a QoL change, meant to make identifying the post easier - currently no block smoke testing flows generate titles, so it's all "Untitled".

dpasque replied:
Nice, great idea!

dpasque commented on packages/calypso-e2e/src/lib/blocks/block-flows/markdown.ts:
```diff
@@ -0,0 +1,66 @@
+	 * test execution.
+	 */
+	async configure( context: EditorContext ): Promise< void > {
+		const editorCanvas = await context.editorPage.getEditorCanvas();
+		const block = editorCanvas.getByRole( 'document', {
+			name: `Block: ${ this.blockSidebarName }`,
+		} );
```
**Nit (non-blocking)**: As an FYI -- there's now a locator that points to the block parent element already in the `EditorContext` object, so you should be able to just use that. 🙂

worldomonation replied:
Gotcha, thanks for the heads up - I replaced both the Markdown and Related Posts flows with the Locator in the context.

dpasque reviewed (approved):
One small comment about a change i made to the `EditorContext`, otherwise LGTM! 🚀

worldomonation committed:
packages/calypso-e2e/src/lib/blocks/block-flows/related-posts.ts
packages/calypso-e2e/src/lib/blocks/block-flows/markdown.ts
- use the locator in the context.

worldomonation committed:
test/e2e/specs/blocks/blocks__jetpack-other.ts
- close settings sidebar

worldomonation committed:
test/e2e/specs/blocks/blocks__jetpack-other.ts
- enable test

worldomonation committed:
packages/calypso-e2e/src/lib/blocks/block-flows/related-posts.ts
- remove flaky logic from Related Posts block flow.
